src/components/LayersPanel/LayersPanel.js

```diff
@@ -27,6 +27,23 @@ class LayersPanel extends React.PureComponent {
       window.docViewer.updateView();
     }
   }
+  componentDidMount() {
+    core.addEventListener('layersUpdated', this.onLayersUpdated.bind(this));
+  }
+
+  componentWillUnmount() {
+    core.removeEventListener('layersUpdated', this.onLayersUpdated.bind(this));
```

Comment:
It looks like this depends on a WebViewer PR, so let me know if  this is ready to merge.